tests/serviceintegration_test.go

```diff
@@ -53,24 +53,24 @@ func TestServiceIntegrationClickhousePostgreSQL(t *testing.T) {
 
 	// THEN
 	// Validates Clickhouse
-	chAvn, err := avnClient.Services.Get(ctx, cfg.Project, chName)
+	chAvn, err := avnGen.ServiceGet(ctx, cfg.Project, chName)
 	require.NoError(t, err)
-	assert.Equal(t, chAvn.Name, ch.GetName())
+	assert.Equal(t, chAvn.ServiceName, ch.GetName())
 	assert.Contains(t, serviceRunningStatesAiven, chAvn.State)
 	assert.Equal(t, chAvn.Plan, ch.Spec.Plan)
 	assert.Equal(t, chAvn.CloudName, ch.Spec.CloudName)
-	assert.Equal(t, chAvn.MaintenanceWindow.DayOfWeek, ch.Spec.MaintenanceWindowDow)
-	assert.Equal(t, chAvn.MaintenanceWindow.TimeOfDay, ch.Spec.MaintenanceWindowTime)
+	assert.Equal(t, string(chAvn.Maintenance.Dow), ch.Spec.MaintenanceWindowDow)
+	assert.Equal(t, chAvn.Maintenance.Time, ch.Spec.MaintenanceWindowTime)
 
 	// Validates PostgreSQL
-	pgAvn, err := avnClient.Services.Get(ctx, cfg.Project, pgName)
+	pgAvn, err := avnGen.ServiceGet(ctx, cfg.Project, pgName)
 	require.NoError(t, err)
-	assert.Equal(t, pgAvn.Name, pg.GetName())
+	assert.Equal(t, pgAvn.ServiceName, pg.GetName())
 	assert.Contains(t, serviceRunningStatesAiven, pgAvn.State)
 	assert.Equal(t, pgAvn.Plan, pg.Spec.Plan)
 	assert.Equal(t, pgAvn.CloudName, pg.Spec.CloudName)
-	assert.Equal(t, pgAvn.MaintenanceWindow.DayOfWeek, pg.Spec.MaintenanceWindowDow)
-	assert.Equal(t, pgAvn.MaintenanceWindow.TimeOfDay, pg.Spec.MaintenanceWindowTime)
+	assert.Equal(t, string(pgAvn.Maintenance.Dow), pg.Spec.MaintenanceWindowDow)
```

Comment:
Since you have already used the state type, it might be helpful to do the same story here. Or would that make too much changes?

Reply:
This has a cascading effect on changes required. Let's do it separately